### ISSUE TITLE
Try mean encoding for the labels

### ISSUE THREAD

**Issue by marco-c** [mozilla/bugbug#71]
It's supposed to be particularly effective for gradient boosted trees, which we are using.

**Comment by saurabhdaalia** [comment]
@marco-c , to clarify, currently we are using Labels as 1 or 0 only. 
Lets say for bug model, if the class is True we encode it as 1 else 0.

Mean Encoding for Label A = [Number of True targets for Class A]/[Total Values for Class A] 
Is this the right way to go?

**Comment by marco-c** [comment]
Not exactly, what we need to try is mean encoding the categorical features, using the labels data.

https://www.coursera.org/lecture/competitive-data-science/concept-of-mean-encoding-b5Gxv (or you can find other resources online) could help understand more.

**Comment by amaaniqbal** [comment]
I wish to work on this. But before that can you please explain about the dataset organizations or which all datasets inside `label/` directory need this?

**Comment by marco-c** [comment]
@amaaniqbal what would you like me to explain exactly?

**Comment by amaaniqbal** [comment]
I meant mean encoding is done on categorical independent variables to make it a good predictor of the dependent variable. So are we supposed to perform it on a dependent variable(target label) here? 

If not then I can only see bug_id in the CSV file except the target variable. Are we scraping independent variables from Bugzilla?

**Comment by marco-c** [comment]
Yes, we are scraping them from Bugzilla, they are in the data/ directory. I think before working on this you'll need to familiarize yourself a bit with the project. The first step would be to read the documentation (README file), especially the last paragraph about the structure of the project.

**Comment by amaaniqbal** [comment]
Yeah, I have gone through it quickly and current requirements are being downloaded for me. But I can't find any data directory anywhere in this repo.

**Comment by marco-c** [comment]
The data will be downloaded automatically when you run the training the first time.

**Comment by amaaniqbal** [comment]
Thanks, I am downloading and setting everything up on a new Linux VM. I will get back right after its setup correctly.

**Comment by ayush1999** [comment]
@marco-c I'm currently experimenting with TargetEncoder, should this option be added to run.py, or should we just manually test the models with this encoder? (Similar to what is being done in #215 )

**Comment by marco-c** [comment]
Let's manually test it first, then we can consider adding it if it's better.

**Comment by ayush1999** [comment]
Sure. Just to be sure, for the bug model, instead of having the labels as 1 for `"bug"` and 0 for `"no_bug"`, we'll be using targetencoders, right?

**Comment by amaaniqbal** [comment]
I don't think you need to modify the labels. You just need to replace the categorical data with mean encoded data. I was a beginner to this repo and had started to work on this. But you can go ahead if you like.

Please refer to https://www.coursera.org/lecture/competitive-data-science/concept-of-mean-encoding-b5Gxv as mentioned by Marco. Its a good reference for this issue.
